packages/apollo/src/commands/service/delete.ts

```diff
@@ -34,9 +34,9 @@ export default class ServiceDelete extends ProjectCommand {
             throw new Error("No service found to link to Engine");
           }
 
-          if (!flags.federated) {
+          if (!flags.federated && !flags.serviceName) {
```

Comment:
Can we remove the logical dependency on `flags.federated` altogether here?

```suggestion
          if (!flags.serviceName) {
```